Kestrel nsrdb fnames

tobin-ford opened the pull request:
## Describe your changes

- Sort returned NSRDB TMY fnames and choose the newest year. 
- Add attribute `kestrel_nsrdb_fnames` to weather_ds returned by `pvdeg.weather.get(..., geospatial=True) which contains the list of files used for traceability.

## Issue ticket number and link
Fixes #259 

Example TMY weather dataset loaded with new `kestrel_nsrdb_fnames` attribute
```
<xarray.Dataset> Size: 1TB
Dimensions:            (time: 8760, gid: 2018267)
Coordinates:
  * gid                (gid) int64 16MB 0 1 2 3 ... 2018264 2018265 2018266
  * time               (time) datetime64[ns] 70kB 2024-01-01T00:30:00 ... 202...
Data variables:
    temp_air           (time, gid) float64 141GB dask.array<chunksize=(8760, 500), meta=np.ndarray>
    wind_speed         (time, gid) float64 141GB dask.array<chunksize=(8760, 500), meta=np.ndarray>
    wind_direction     (time, gid) float64 141GB dask.array<chunksize=(8760, 500), meta=np.ndarray>
    dhi                (time, gid) float64 141GB dask.array<chunksize=(8760, 500), meta=np.ndarray>
    ghi                (time, gid) float64 141GB dask.array<chunksize=(8760, 500), meta=np.ndarray>
    dni                (time, gid) float64 141GB dask.array<chunksize=(8760, 500), meta=np.ndarray>
    relative_humidity  (time, gid) float64 141GB dask.array<chunksize=(8760, 500), meta=np.ndarray>
    albedo             (time, gid) float64 141GB dask.array<chunksize=(8760, 500), meta=np.ndarray>
Attributes:
    package:               nsrdb
    version:               4.1.2.dev4+g3b38bc8.d20250228
    version_record:        {"nsrdb": "4.1.2.dev4+g3b38bc8.d20250228", "farms"...
    kestrel_nsrdb_fnames:  ['/datasets/NSRDB/current/nsrdb_tmy-2024.h5']
```

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Code changes are covered by tests.
- [x] API.rst is up to date, along with other sphinx docs pages
- [ ] Example notebooks are rerun and differences in results scrutinized
- [x] What's new changelog has been updated in the docs

RDaxini reviewed (approved):
LGTM just one comment on the linting, and I'd also suggest adding a quick whatsnew entry but that's not major

RDaxini reviewed (commented):
One other comment... could we add some tests? at least to check the sorting, that should be straight forward.

tobin-ford commented:
@RDaxini I've added tests for sorting and the following cases for `weather.get_NSRDB`
tmy: take the most recent tmy entry, tmy outputs are a single h5 .  
year: take all files in the directory, yearly outputs are split into many h5s. 

I updated the whatsnew entry for v0.6.0. I'll merge once the actions end. Thanks for the review!
